pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -673,9 +689,125 @@ func addUserConfig(config map[string]interface{}, controlPlane *rkev1.RKEControl
+// interface slice and converting each entry to a string using Sprintf.
+func convertInterfaceSliceToStringSlice(input []interface{}) []string {
+	var stringArr []string
+	for _, v := range input {
+		stringArr = append(stringArr, v.(string))
+	}
+	return stringArr
+}
+
+// appendToInterface will return an interface that has the value appended to it. The interface returned will always be
+// a slice of strings, and will convert a raw string to a slice of strings.
+func appendToInterface(input interface{}, elem string) []string {
+	switch input := input.(type) {
+	case []interface{}:
+		stringArr := convertInterfaceSliceToStringSlice(input)
+		return appendToInterface(stringArr, elem)
+	case []string:
+		return append(input, elem)
+	case string:
+		return []string{input, elem}
+	}
+	return []string{elem}
+}
+
+// convertInterfaceToStringSlice converts an input interface to a string slice by determining its type and converting
+// it accordingly. If it is not a known convertible type, an empty string slice is returned.
+func convertInterfaceToStringSlice(input interface{}) []string {
+	switch input := input.(type) {
+	case []interface{}:
+		return convertInterfaceSliceToStringSlice(input)
+	case []string:
+		return input
+	case string:
+		return []string{input}
+	}
+	return []string{}
+}
+
+// renderArgAndMount takes the value of the existing value of the argument and mount and renders an output argument and
+// mount based on the value of the input interfaces. It will always return a set of slice of strings.
+func renderArgAndMount(existingArg interface{}, existingMount interface{}, runtime string, defaultSecurePort string, defaultCertDir string) ([]string, []string) {
+	retArg := convertInterfaceToStringSlice(existingArg)
+	retMount := convertInterfaceToStringSlice(existingMount)
+	renderedCertDir := fmt.Sprintf(defaultCertDir, runtime)
+	certDirMount := fmt.Sprintf("%s:%s", renderedCertDir, renderedCertDir)
+	certDirArg := fmt.Sprintf("%s=%s", CertDirArgument, renderedCertDir)
+	securePortArg := fmt.Sprintf("%s=%s", SecurePortArgument, defaultSecurePort)
+	if len(retArg) > 0 {
+		tlsCF := getArgValue(retArg, TLSCertFileArgument, "=")
+		if tlsCF == "" {
+			certDir := getArgValue(retArg, CertDirArgument, "=")
+			if certDir != "" {
+				certDirArg = ""
+				certDirMount = fmt.Sprintf("%s:%s", certDir, certDir)
+			}
+		} else {
+			certDirArg = ""
+			dir := filepath.Dir(tlsCF)
+			certDirMount = fmt.Sprintf("%s:%s", dir, dir)
+		}
+		sPA := getArgValue(retArg, SecurePortArgument, "=")
+		if sPA != "" {
+			securePortArg = ""
+		}
+	}
+	if certDirArg != "" {
+		retArg = appendToInterface(existingArg, certDirArg)
+	}
+	if securePortArg != "" {
+		retArg = appendToInterface(retArg, securePortArg)
+	}
+	if runtime == rancherruntime.RuntimeRKE2 {
+		retMount = appendToInterface(existingMount, certDirMount)
+	}
+	return retArg, retMount
```

Comment:
Comments in here would be good. Stuff like: "the user provided this setting, so we set this setting" etc.

Reply:
Verbose comments added, good idea!